### PR TITLE
[terra-notification-dialog] Fixed A11y issues

### PR DESCRIPTION
**Pull request by saket2403**
### Summary

**What was changed:**
- Removed the 'BeginModalDialog' and 'EndModalDialog' messages from the ModalContent when it is called from the NotificationDialog.
- The title was read twice by the screen readers. I've changed the ariaLabel, ariaLabelledBy and ariaDescribedBy to properly read the content of the NotificationDialog.

**Why it was changed:**

To Fix Accessibility issues as discussed with UX

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9855 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-abstract-modal/src/AbstractModal.jsx [hunk 1]
```diff
@@ -65,6 +73,10 @@ const propTypes = {
    * Callback function to set the reference of the element that will receive focus when the Slide content is visible.
    */
   setModalFocusElementRef: PropTypes.func,
+  /**
+   * If set to true, the AbstractModal is rendered inside a NotificationDialog.
+   */
+  isCalledFromNotificationDialog: PropTypes.bool,
```
Comment (supreethmr) [inline packages/terra-abstract-modal/src/AbstractModal.jsx:79]: let's make this prop private, As it's used internally on notification-dialog

Comment (supreethmr) [inline packages/terra-abstract-modal/src/AbstractModal.jsx:79]: I am not able to find any references for `isCalledFromNotificationDialog` prop

File: packages/terra-notification-dialog/src/NotificationDialog.jsx [hunk 2]
```diff
@@ -156,16 +156,17 @@ const NotificationDialog = (props) => {
   /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
   return (
     <AbstractModal
-      ariaLabel={signalWord}
-      aria-labelledby="notification-dialog-signal-word"
-      aria-describedby={dialogTitle ? 'notification-dialog-title' : 'notification-dialog-signal-word'}
-      role="alertdialog"
+      ariaLabel=""
```
Comment (sdadn) [inline packages/terra-notification-dialog/src/NotificationDialog.jsx:159]: Why is this needed? Why not remove the prop?

File: packages/terra-abstract-modal/src/AbstractModal.jsx (hunk 1 above)
Comment (saket2403) [reply packages/terra-abstract-modal/src/AbstractModal.jsx:80]: Done [71cf564](https://github.com/cerner/terra-framework/pull/1894/commits/71cf564a7392da594d93aead47c597ef34f9a613)

Comment (saket2403) [reply packages/terra-abstract-modal/src/AbstractModal.jsx:80]: Done [71cf564](https://github.com/cerner/terra-framework/pull/1894/commits/71cf564a7392da594d93aead47c597ef34f9a613)

File: packages/terra-notification-dialog/src/NotificationDialog.jsx (hunk 2 above)
Comment (saket2403) [reply packages/terra-notification-dialog/src/NotificationDialog.jsx:159]: `ariaLabel` is marked as required in AbstractModal. It gives a console error if not passed. So passing an empty string prevents that error and the screen reader announcements are taken care of by other aria attributes

File: packages/terra-abstract-modal/src/AbstractModal.jsx (hunk 1 above)
Comment (supreethmr) [reply packages/terra-abstract-modal/src/AbstractModal.jsx:80]: since it is boolean, Can we get this prop assigned with default value `false` for consistency

File: packages/terra-abstract-modal/src/_ModalContent.jsx [hunk 3]
```diff
@@ -138,38 +154,44 @@ const ModalContent = forwardRef((props, ref) => {
...
-              let useText = text;
-              if (Array.isArray(text)) {
-                useText = text.join('');
-              }
-              return (
-                <VisuallyHiddenText text={useText} />
-              );
-            }}
-          </FormattedMessage>
+          {(!isCalledFromNotificationDialog && isCalledFromNotificationDialog !== undefined) && (
```
Comment (supreethmr) [inline packages/terra-abstract-modal/src/_ModalContent.jsx:166]: with default value false, you will able to reduce the above condition to 
```suggestion
          {(!isCalledFromNotificationDialog) && (
```

Comment (saket2403) [reply packages/terra-abstract-modal/src/_ModalContent.jsx]: Done [5df6594](https://github.com/cerner/terra-framework/pull/1894/commits/5df65942439721cd6bd96dc4dc96ad8e7ff4acf2)

**Review by supreethmr — APPROVED** [review]

**Review by sugan2416 — APPROVED** [review]

File: packages/terra-notification-dialog/src/NotificationDialog.jsx [hunk 4]
```diff
@@ -156,16 +156,15 @@ const NotificationDialog = (props) => {
   /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
   return (
     <AbstractModal
-      ariaLabel={signalWord}
-      aria-labelledby="notification-dialog-signal-word"
-      aria-describedby={dialogTitle ? 'notification-dialog-title' : 'notification-dialog-signal-word'}
-      role="alertdialog"
+      ariaLabelledBy="notification-dialog-signal-word"
+      role={signalWord === 'Alert' ? 'alertdialog' : 'dialog'}
```
Comment (supreethmr) [inline packages/terra-notification-dialog/src/NotificationDialog.jsx:160]: this should not be dependent on signalWord instead we should use `variant` dialog needs to added only for custom variant

Comment (saket2403) [reply packages/terra-notification-dialog/src/NotificationDialog.jsx]: We will keep that for the other a11y changes. Reverted this to alertdialog

File: packages/terra-abstract-modal/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * Fixed title getting announced twice.
```
Comment (sugan2416) [inline packages/terra-abstract-modal/CHANGELOG.md:6]: Do we not want to mention about addition of two new aria props?

Comment (saket2403) [reply packages/terra-abstract-modal/CHANGELOG.md:6]: Abstract modal is deprecated. So its ok since consumers will not be using this directly instead they should be using modal mamanger.

File: packages/terra-notification-dialog/src/NotificationDialog.jsx [hunk 6]
```diff
@@ -156,16 +156,16 @@ const NotificationDialog = (props) => {
...
-      aria-describedby={dialogTitle ? 'notification-dialog-title' : 'notification-dialog-signal-word'}
+      ariaLabelledBy="notification-dialog-signal-word"
+      aria-describedby={dialogTitle ? 'notification-dialog-title' : undefined}
       role="alertdialog"
       classNameModal={classNames(cx('notification-dialog', theme.className), customProps.className)}
-      isOpen
       onRequestClose={() => {}}
-      closeOnEsc={false}
+      isOpen
+      closeOnEsc
```
Comment (sugan2416) [inline packages/terra-notification-dialog/src/NotificationDialog.jsx:165]: Do we still need this, since its true by default in AbstractModal ?

Comment (saket2403) [reply packages/terra-notification-dialog/src/NotificationDialog.jsx]: Removed

**Review by supreethmr — APPROVED** [review]
+1 for screen reader changes

**Comment by rahulkumar8cs** [comment]
+1
